universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java

```diff
@@ -164,4 +161,35 @@ private Result withMessage(Result result, String message) {
     }
     return result;
   }
+
+  private QuestionForm createQuestionForm(Request request, String type)
+      throws InvalidQuestionTypeException {
+    QuestionType questionType;
+    try {
+      questionType = QuestionType.of(type);
+    } catch (InvalidQuestionTypeException e) {
+      throw new InvalidQuestionTypeException(
+          String.format(
+              "unrecognized question type: '%s', accepted values include: %s",
+              type.toUpperCase(), Arrays.toString(QuestionType.values())));
```

Comment:
praise: really nice error message!

Reply:
😄 This was already here- just moved it around.